Source/OxyPlot/Series/BarSeries/LinearBarSeries.cs

```diff
@@ -0,0 +1,333 @@
+        /// </summary>
+        private readonly List<int> rectanglesPointIndexes = new List<int>();
+
+        /// <summary>
+        /// The default color.
+        /// </summary>
+        private OxyColor defaultColor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LinearBarSeries" /> class.
+        /// </summary>
+        public LinearBarSeries()
+        {
+            this.FillColor = OxyColors.Automatic;
+            this.BarWidth = 5;
+            this.StrokeColor = OxyColors.Black;
+            this.StrokeThickness = 0;
+            this.TrackerFormatString = XYAxisSeries.DefaultTrackerFormatString;
+            this.NegativeFillColor = OxyColors.Undefined;
+            this.NegativeStrokeColor = OxyColors.Undefined;
+        }
+
+        /// <summary>
+        /// Gets or sets the color of the interior of the bars.
+        /// </summary>
+        /// <value>The color.</value>
+        public OxyColor FillColor { get; set; }
+
+        /// <summary>
+        /// Gets or sets the width of the bars.
+        /// </summary>
+        /// <value>The width of the bars.</value>
+        public double BarWidth { get; set; }
+
+        /// <summary>
+        /// Gets or sets the thickness of the curve.
+        /// </summary>
+        /// <value> The stroke thickness.</value>
+        public double StrokeThickness { get; set; }
+
+        /// <summary>
+        /// Gets or sets the color of the border around the bars.
+        /// </summary>
+        /// <value>The color of the stroke.</value>
+        public OxyColor StrokeColor { get; set; }
+
+        /// <summary>
+        /// Gets or sets the color of the interior of the bars when the value is negative.
+        /// </summary>
+        /// <value>The color.</value>
+        public OxyColor NegativeFillColor { get; set; }
+
+        /// <summary>
+        /// Gets or sets the color of the border around the bars when the value is negative.
+        /// </summary>
+        /// <value>The color of the stroke.</value>
+        public OxyColor NegativeStrokeColor { get; set; }
+
+        /// <summary>
+        /// Gets the actual color.
+        /// </summary>
+        /// <value>The actual color.</value>
+        public OxyColor ActualColor
+        {
+            get
+            {
+                return this.FillColor.GetActualColor(this.defaultColor);
+            }
+        }
+
+        /// <summary>
+        /// Gets the nearest point.
+        /// </summary>
+        /// <param name="point">The point.</param>
+        /// <param name="interpolate">interpolate if set to <c>true</c> .</param>
+        /// <returns>A TrackerHitResult for the current hit.</returns>
+        public override TrackerHitResult GetNearestPoint(ScreenPoint point, bool interpolate)
+        {
+            var rectangleIndex = this.rectangles.BinarySearch(0, this.rectangles.Count, new OxyRect(), new NearestPointComparer(point));
```

Comment:
is there a binary search extension method that takes a Func? I think that would be cleaner than creating an `IComparer<OxyRect>` class.